src/test/java/io/vavr/control/ValidationTest.java

```diff
@@ -300,6 +300,20 @@ public void shouldConvertFailureToU() {
         assertThat(result).isEqualTo(3);
     }
 
+
+    // getOrElseThrow(Function)
+
+    @Test
+    public void shouldReturnValidWhenGetOrElseThrowWithFunctionOnValid() {
+      final Integer actual = Validation.<String, Integer> valid(1).getOrElseThrow(s -> new RuntimeException(s));
+      assertThat(actual).isEqualTo(1);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void shouldThrowWhenGetOrElseThrowWithFunctionOnInvalid() {
+      Validation.<String, Integer> invalid("some error").getOrElseThrow(s -> new RuntimeException(s));
+    }
+
```

Comment:
We may also consider to add a test for an edge case, where `null` provided instead of `exceptionFunction`.

Reply:
Something like 
```
@Test(expected = NullPointerException.class)
public void shouldThrowNullPointerExceptionWhenExceptionFunctionIsNullOnValid() {
    Validation.<String, Integer> valid(1).getOrElseThrow(null);
}

@Test(expected = NullPointerException.class)
public void shouldThrowNullPointerExceptionWhenExceptionFunctionIsNullOnInvalid() {
    Validation.<String, Integer> invalid("some error").getOrElseThrow(null);
}
```

Reply:
Done (I adapted the test names a bit, and added types to make it compile)

Reply:
Thank you @sir4ur0n !
I believe we need @danieldietrich review and apporoval.